src/task_scl_eff_pot_hab.py

```diff
@@ -3,180 +3,200 @@
+            .gte(self.thresholds["patch_size"])
+            .selfMask()
+            .reproject(crs="EPSG:4326", scale=self.scale)
+        )
+        current_range = (
+            self.extirpated.updateMask(self.historic_range.eq(0)).eq(1).selfMask()
+        )
 
-foo = SCLPolygons.delineate_polygons(SCLPolygons)
-print(foo.getInfo())
+        connected_structural_habitat = str_hab_patch.updateMask(current_range)
+        potential_habitat = connected_structural_habitat.updateMask(
+            self.hii.lt(self.thresholds["hii"])
+        ).selfMask()
+        excluded_habitat = connected_structural_habitat.updateMask(
+            self.hii.gte(self.thresholds["hii"])
+        ).selfMask()
+        effective_potential_habitat = potential_habitat.addBands(
+            excluded_habitat
+        ).rename(["eff_pot_hab", "excl_hab"])
+
+        # probability_calc = self.elevation.expression(
+        #     "1 - (2.71828 ** (-(2.71828 ** (- 0.069 - 0.044 * ((dem - 639) / 592) - 0.12 * ((hii - 7.72) / 5.47)))))",
+        #     {"dem": self.elevation, "hii": self.hii},
+        # )
+
+        high_probability = self.probability.gte(self.thresholds["probability"]).multiply(
+            2
+        )
 
+        low_probability = self.probability.lt(self.thresholds["probability"]).multiply(
+            1
+        )
 
+        probability_calc = high_probability.add(low_probability).selfMask()
 
+        hii_sum = (
+            effective_potential_habitat.select(0)
+            .unmask(0)
+            .add(effective_potential_habitat.select(1).unmask(0).multiply(2))
+            .selfMask()
+            .remap([1, 2], [2, 3])
+        )
 
-    def calc(self):
-        self.delineate_polygons()
+        hii_ecoregion = self.ecoregion_country.add(hii_sum)
+
+        minPatchSize = self.density.remap(
+            self.remap_lists["density"], self.remap_lists["min_patch_size"]
+        ).updateMask(hii_ecoregion)
+        minStpStnSize = self.density.remap(
+            self.remap_lists["density"], self.remap_lists["min_stepping_stone_size"]
+        ).updateMask(hii_ecoregion)
+
+        hii_grp = hii_ecoregion.addBands(
+            [
+                probability_calc,
+                hii_sum,
+                ee.Image.pixelArea().multiply(0.000001),
+                minPatchSize,
+                minStpStnSize,
+            ]
+        ).reproject(crs="EPSG:4326", scale=self.scale)
+
+        polygonReducer = (
+            ee.Reducer.mode()
+            .combine(ee.Reducer.mode().unweighted(), "hii_")
+            .combine(ee.Reducer.sum().unweighted(), "area_")
+            .combine(ee.Reducer.mode().unweighted(), "patch_")
+            .combine(ee.Reducer.mode().unweighted(), "stpStn_")
+        )
+        hii_grp_poly = hii_grp.reduceToVectors(
+            reducer=polygonReducer,
+            geometry=ee.Geometry.MultiPolygon(self.aoi).bounds(),
+            crs=self.crs,
+            scale=self.scale,
+            maxPixels=1e13,
```

Comment:
There's a `max_pixels` property in task_base as well, so you can use that like you are crs/scale

Reply:
changed to self.ee_max_pixels